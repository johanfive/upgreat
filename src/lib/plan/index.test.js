const Plan = require('./')

const fs = require('../../util/fs')
jest.mock('../../util/fs')

const createPlan = require('./createPlan')
jest.mock('./createPlan')

describe('run', () => {
  it('should error if path is wrong', async () => {
    const plan = new Plan([])

    const logSpy = jest.spyOn(plan, 'log')
    logSpy.mockImplementation(() => {})
    const errorSpy = jest.spyOn(plan, 'error')
    errorSpy.mockImplementation(err => {
      throw new Error(err)
    })
    fs.readFile.mockImplementation(() => Promise.reject())

    await expect(plan.run()).rejects.toMatchSnapshot()
  })

  it('should greet', async () => {
    const plan = new Plan([])

    const spy = jest.spyOn(plan, 'log')
    spy.mockImplementation(() => {})
    fs.readFile.mockImplementation(() => {
      return '{"something": "yes"}'
    })
    createPlan.mockImplementation(() => ({
      plan: [{ diff: 'major' }, {}],
      missingPeers: ['babel-plugin-long-name-transformer', 'lodash'],
      errors: [
        { name: 'some-pkg', err: 404 },
        { name: 'other-pkg', err: 'exotic' },
      ],
    }))
    fs.mkUpgreatDir.mockImplementation(() => Promise.resolve())
    fs.writeFilePrettyJson.mockImplementation(() => Promise.resolve())

    await plan.run()

    expect(spy.mock.calls).toMatchSnapshot()
  })
})
