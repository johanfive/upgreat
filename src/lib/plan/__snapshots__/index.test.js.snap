// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`run should error if path is wrong 1`] = `[Error: could not read package.json, is the repo path correct?]`;

exports[`run should greet 1`] = `
Array [
  Array [
    "ðŸ‘‹ hi, will read package.json",
  ],
  Array [
    "you have missing peerDependencies that you will need",
  ],
  Array [
    "  babel-plugin-long-name-transformer,lodash",
  ],
  Array [
    "unable to retrieve version info for some packages, these will be ignored in the upgrade plan",
  ],
  Array [
    "  some-pkg: 404
  other-pkg: exotic",
  ],
  Array [
    "ðŸ—’  wrote upgrade plan to .upgreat/plan.json",
  ],
  Array [
    "run upgreat up to start the upgrades!",
  ],
]
`;
